### ISSUE TITLE
PlantEquipmentOperation:ThermalEnergyStorage should warn of incorrect component operation type

### ISSUE THREAD

**Issue by mjwitte** [NREL/EnergyPlus#6958]
Issue overview
--------------
User file failed to charge ice storage because the last field in this object was incorrect:
```
PlantEquipmentOperation:ThermalEnergyStorage,
    CHW Loop Scheme 1,       !- Name
    Ice Thermal Storage On-peak,  !- On-Peak Schedule
    Ice Thermal Storage Charging,  !- Charging Availability Schedule
    7.00,                    !- Non-Charging Chilled Water Temperature {C}
    -5.00,                   !- Charging Chilled Water Temperature {C}
    Chiller:Electric:EIR,    !- Component 1 Object Type
    Chiller,                 !- Component 1 Name
    Primary CHW Loop Pump Water Outlet Node,  !- Component 1 Demand Calculation Node Name
    Chiller CHW Outlet Node, !- Component 1 Setpoint Node Name
    0.001351,                !- Component 1 Flow Rate {m3/s}
    Cooling,                 !- Component 1 Operation Type
    ThermalStorage:Ice:Detailed,  !- Component 2 Object Type
    Ice Thermal Storage,     !- Component 2 Name
    Chiller CHW Outlet Node, !- Component 2 Demand Calculation Node Name
    Ice Thermal Storage Water Outlet Node,  !- Component 2 Setpoint Node Name
    autosize,                !- Component 2 Flow Rate {m3/s}
    Cooling;                 !- Component 2 Operation Type
```
The I/O Reference states this should be "Dual":
Field: Component <#> Operation Type
This alpha field specifies the operation mode for the equipment. The equipment can be in any of the three modes viz. Cooling, Heating and Dual. Dual is used when the components both as heating and cooling equipment (for example heat pumps). *Ice storage units can potentially either heat or cool the circulating fluid and thus should be defined as Dual mode.*

Seems the input processing for this object could detect ice storage and correct this with a warning.

### Details
Some additional details for this issue (if relevant):
 - Platform Win64
 - Version of EnergyPlus Version 9.0.0-771c3799ec
 - Helpdesk ticket number 12685

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Defect file added EnergyPlusDevSupport\DefectFiles
 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

**Comment by jmarrec** [comment]
I couldn't figure out why my unit test kept crashing.

I'm pretty sure I uncovered another problem: https://github.com/jmarrec/EnergyPlus/blob/5ddf585d0378775ebe219927560b332b6ed68d82/src/EnergyPlus/SetPointManager.cc#L8872

That's probably a left over from when it was using internally created SPM:Scheduled for TES control.

**Comment by mjwitte** [comment]
If I'm following correctly, that would just put the wrong name into `AllSetPtMgr` which would only affect a duplicate setpoint node error message. But, certainly it should be fixed.

**Comment by jmarrec** [comment]
It would crash if you don't have at least the same amount of SPM:Scheduled than the number of internally generated TES SPMs. I fixed it in #7143
